pyembed/src/python_resource_collector.rs

```diff
@@ -314,3 +341,29 @@ impl OxidizedResourceCollector {
             .into_object())
     }
 }
+
+#[cfg(test)]
+mod test_pytempdir {
+    use super::PyTempDir;
+
+    fn get_py<'interp, 'rsrc, 'py>() -> (crate::MainPythonInterpreter<'py, 'interp, 'rsrc>, cpython::Python<'py>) {
+        let mut config = crate::OxidizedPythonInterpreterConfig::default();
+        config.interpreter_config.parse_argv = Some(false);
+        config.set_missing_path_configuration = false;
+        let mut interp = crate::MainPythonInterpreter::new(config).unwrap();
+        let py = interp.acquire_gil().unwrap();
+        (interp, py)
+    }
+
+    #[test]
+    fn lifetimes() {
+        let path = {
+            let (_interp, py) = get_py();
+            let temp_dir = PyTempDir::new(py).unwrap();
+            drop(py);  // PyTempDir::drop reacquires the GIL for itself
+            assert!(temp_dir.path().is_dir());
+            temp_dir.path().to_path_buf()
+        };
+        assert!(!path.is_dir());
+    }
```

Comment:
I'll likely refactor this in flight to use `rusty_fork_test!`. The reason is that I've run into many issues where Python interpreter state in the current process gets "corrupted" and causes crashes. It is much safer to isolate each test spinning up a Python interpreter to its own process.